keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,636 @@
+        metrics=None,
+        **kwargs,
+    ):
+        """Compiles the YOLOV8Detector.
+
+        `compile()` mirrors the standard Keras `compile()` method, but has one
+        key distinction -- two losses must be provided: `box_loss` and
+        `classification_loss`.
+
+        Args:
+            box_loss: a Keras loss to use for box offset regression. A
+                preconfigured loss is provided when the string "iou" is passed.
+            classification_loss: a Keras loss to use for box classification. A
+                preconfigured loss is provided when the string
+                "binary_crossentropy" is passed.
+            box_loss_weight: (optional) float, a scaling factor for the box
+                loss. Defaults to 7.5.
+            classification_loss_weight: (optional) float, a scaling factor for
+                the classification loss. Defaults to 0.5.
+            kwargs: most other `keras.Model.compile()` arguments are supported
+                and propagated to the `keras.Model` class.
+        """
+        if metrics is not None:
+            raise ValueError("User metrics not yet supported for YOLOV8")
+
+        if isinstance(box_loss, str):
+            if box_loss == "iou":
+                box_loss = YOLOV8IoULoss(reduction="sum")
+            else:
+                raise ValueError(
+                    f"Invalid box loss for YOLOV8Detector: {box_loss}. Box "
+                    "loss should be a keras.Loss or the string 'iou'."
+                )
+        if isinstance(classification_loss, str):
+            if classification_loss == "binary_crossentropy":
+                classification_loss = keras.losses.BinaryCrossentropy(
+                    reduction="sum"
+                )
+            else:
+                raise ValueError(
+                    "Invalid classification loss for YOLOV8Detector: "
+                    f"{classification_loss}. Classification loss should be a "
+                    "keras.Loss or the string 'binary_crossentropy'."
+                )
+
+        self.box_loss = box_loss
+        self.classification_loss = classification_loss
+        self.box_loss_weight = box_loss_weight
+        self.classification_loss_weight = classification_loss_weight
+
+        losses = {
+            "box": self.box_loss,
+            "class": self.classification_loss,
+        }
+
+        super().compile(loss=losses, **kwargs)
+
+    def train_step(self, data):
+        x, y = unpack_input(data)
+
+        with tf.GradientTape() as tape:
+            outputs = self(x, training=True)
+            box_pred, cls_pred = outputs["boxes"], outputs["classes"]
+            total_loss = self.compute_loss(x, y, box_pred, cls_pred)
+
+        trainable_vars = self.trainable_variables
+
+        gradients = tape.gradient(total_loss, trainable_vars)
+        self.optimizer.apply_gradients(zip(gradients, trainable_vars))
+
+        return super().compute_metrics(x, {}, {}, sample_weight={})
```

Comment:
looks like you don't support COCO metrics here; additionally there is no early exit clause for when no metrics are included.  

Any difficulties in mirroring the RetinaNet behavior?  https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/object_detection/retina_net/retina_net.py

Reply:
https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/object_detection/retina_net/retina_net.py#L477

Reply:
This is intentional -- you'll notice in `compile` that I throw if metrics are included.

I am leaving this as a follow-up task.